src/ripple/core/SQLInterface.h

```diff
@@ -0,0 +1,674 @@
+    getMinLedgerSeq(SQLDatabase& db, TableType type) = 0;
+
+    virtual boost::optional<LedgerIndex>
+    getMinLedgerSeq(SQLDatabase_* db, TableType type) = 0;
+
+    virtual boost::optional<LedgerIndex>
+    getMaxLedgerSeq(SQLDatabase& db, TableType type) = 0;
+
+    virtual boost::optional<LedgerIndex>
+    getMaxLedgerSeq(SQLDatabase_* db, TableType type) = 0;
+
+    virtual void
+    deleteByLedgerSeq(
+        SQLDatabase& db,
+        TableType type,
+        LedgerIndex ledgerSeq) = 0;
+
+    virtual void
+    deleteByLedgerSeq(
+        SQLDatabase_* db,
+        TableType type,
+        LedgerIndex ledgerSeq) = 0;
+
+    virtual void
+    deleteBeforeLedgerSeq(
+        SQLDatabase& db,
+        TableType type,
+        LedgerIndex ledgerSeq) = 0;
+
+    virtual void
+    deleteBeforeLedgerSeq(
+        SQLDatabase_* db,
+        TableType type,
+        LedgerIndex ledgerSeq) = 0;
+
+    virtual int
+    getRows(SQLDatabase& db, TableType type) = 0;
+
+    virtual int
+    getRows(SQLDatabase_* db, TableType type) = 0;
+
+    virtual std::tuple<int, int, int>
+    getRowsMinMax(SQLDatabase& db, TableType type) = 0;
+
+    virtual std::tuple<int, int, int>
+    getRowsMinMax(SQLDatabase_* db, TableType type) = 0;
+
+    virtual void
+    insertAcquireDBIndex(SQLDatabase& db, std::uint32_t index_) = 0;
+
+    virtual std::pair<bool, boost::optional<std::string>>
+    selectAcquireDBLedgerSeqs(SQLDatabase& db, std::uint32_t index) = 0;
+
+    virtual std::
+        tuple<bool, boost::optional<std::string>, boost::optional<std::string>>
+        selectAcquireDBLedgerSeqsHash(SQLDatabase& db, std::uint32_t index) = 0;
+
+    virtual bool
+    updateLedgerDBs(
+        SQLDatabase& txdb,
+        SQLDatabase& lgrdb,
+        std::shared_ptr<Ledger const> const& ledger,
+        std::uint32_t const index,
+        beast::Journal const j,
+        std::atomic<bool>& stop) = 0;
+
+    virtual void
+    updateAcquireDB(
+        SQLDatabase& db,
+        std::shared_ptr<Ledger const> const& ledger,
+        std::uint32_t const index,
+        std::uint32_t const lastSeq,
+        boost::optional<std::string> seqs) = 0;
+
+    virtual bool
+    saveValidatedLedger(
+        SQLDatabase& ldgDB,
+        SQLDatabase& txnDB,
+        Application& app,
+        std::shared_ptr<Ledger const> const& ledger,
+        bool current) = 0;
+
+    virtual bool
+    saveValidatedLedger(
+        SQLDatabase_* ldgDB,
+        SQLDatabase_* txnDB,
+        Application& app,
+        std::shared_ptr<Ledger const> const& ledger,
+        bool current) = 0;
+
+    virtual bool
+    loadLedgerInfoByIndex(
+        SQLDatabase& db,
+        SQLInterface::SQLLedgerInfo& info,
+        beast::Journal const& j,
+        LedgerIndex ledgerSeq) = 0;
+
+    virtual bool
+    loadLedgerInfoByIndex(
+        SQLDatabase_* db,
+        SQLInterface::SQLLedgerInfo& info,
+        beast::Journal const& j,
+        LedgerIndex ledgerSeq) = 0;
```

Comment:
There are a bunch of functions with the same name, that do the same thing, that differ in one parameter. Can we merge these functions together?